repose-aggregator/components/filters/rackspace-identity-basic-auth/src/main/scala/org/openrepose/filters/rackspaceidentitybasicauth/RackspaceIdentityBasicAuthFilter.scala

```diff
@@ -100,19 +98,34 @@ class RackspaceIdentityBasicAuthFilter @Inject()(configurationService: Configura
   }
 
   override def doFilter(servletRequest: ServletRequest, servletResponse: ServletResponse, filterChain: FilterChain) {
-    new FilterLogicHandlerDelegate(servletRequest, servletResponse, filterChain).doFilter(this)
+    val wrappedHttpRequest = new HttpServletRequestWrapper(servletRequest.asInstanceOf[HttpServletRequest])
+    val httpResponse = servletResponse.asInstanceOf[HttpServletResponse]
+
+    val filterAction = handleRequest(wrappedHttpRequest, httpResponse)
+    filterAction match {
```

Comment:
In a general sense, PASS is the most common action. Of course, it isn't used in this filter.

NOT_SET would have been the default on the FilterDirector, which was the return type from these methods in the old pattern. I guess the idea was that you had to be explicit about the FilterAction. Since that code is now in the filters themselves, it will always be explicit, so we probably don't need NOT_SET anymore.

It's true that FilterAction is an enumeration that lends itself to the old pattern. In fact, that's the reason I've kept using it -- it prevents me from having to refactor all of the handleRequest(...) and handleResponse(...) methods. Even so, if we're not getting rid of mutability this pass, I don't have any real desire to get rid of FilterAction either.

I'll also note that I used @wdschei FilterAction in commons utils rather than the one in core.

So here are a couple of options going forward:
1. Do nothing.
2. Leave code comments indicating that PASS and NOT_SET are not expected to be used. Note that these comments will have to be maintained with the code, which sometimes doesn't happen since the compile doesn't check comments.
3. Drop NOT_SET from FilterAction. I don't know if this is actually possible for the moment, since other code might rely on it. Also, we still don't use PASS in this filter, so that may need to be addressed separately.
4. Only react to expected FilterActions by using an if/else if/else block. This is basically just a match, though, so why?
5. Combine PASS and NOT_SET into a catch-all in the match (i.e., case _ => ...). This seems to indicate that only the other FilterActions are expected, but if the unexpected happens, do something. I'm not sure what the something that we should be. Log and fail with a 500 (or just throw an exception)?